cloud/scope/vpc_cluster.go

```diff
@@ -166,3 +197,281 @@ func (s *VPCClusterScope) Close() error {
+			}
+			return
+		} else if s.NetworkStatus().VPC == nil {
+			s.IBMVPCCluster.Status.Network.VPC = resource
+		}
+		s.NetworkStatus().VPC.Set(*resource)
+	default:
+		s.V(3).Info("unsupported resource type", "resourceType", resourceType)
+	}
+}
+
+// TagResource will attach a user Tag to a resource.
+func (s *VPCClusterScope) TagResource(tagName string, resourceCRN string) error {
+	// Verify the Tag we wish to use exists, otherwise create it.
+	exists, err := s.CheckTagExists(tagName)
+	if err != nil {
+		return fmt.Errorf("failure checking if tag exists: %w", err)
+	}
+
+	// Create tag if it doesn't exist.
+	if !exists {
+		createOptions := &globaltaggingv1.CreateTagOptions{}
+		createOptions.SetTagNames([]string{tagName})
+		if _, _, err := s.GlobalTaggingClient.CreateTag(createOptions); err != nil {
+			return fmt.Errorf("failure creating tag: %w", err)
+		}
+	}
+
+	// Finally, tag resource.
+	tagOptions := &globaltaggingv1.AttachTagOptions{}
+	tagOptions.SetResources([]globaltaggingv1.Resource{
+		{
+			ResourceID: ptr.To(resourceCRN),
+		},
+	})
+	tagOptions.SetTagName(tagName)
+	tagOptions.SetTagType(globaltaggingv1.AttachTagOptionsTagTypeUserConst)
+
+	if _, _, err = s.GlobalTaggingClient.AttachTag(tagOptions); err != nil {
+		return fmt.Errorf("failure tagging resource: %w", err)
+	}
+
+	return nil
+}
+
+// ReconcileVPC reconciles the cluster's VPC.
+func (s *VPCClusterScope) ReconcileVPC() (bool, error) {
+	// If VPC id is set, that indicates the VPC already exists.
+	vpcID, err := s.GetVPCID()
+	if err != nil {
+		return false, fmt.Errorf("failed to retrieve vpc id: %w", err)
+	}
+	if vpcID != nil {
+		s.V(3).Info("VPC id is set", "id", vpcID)
+		vpcDetails, _, err := s.VPCClient.GetVPC(&vpcv1.GetVPCOptions{
+			ID: vpcID,
+		})
+		if err != nil {
+			return false, fmt.Errorf("failed to retrieve vpc by id: %w", err)
+		} else if vpcDetails == nil {
+			return false, fmt.Errorf("failed to retrieve vpc with id: %s", *vpcID)
+		}
+		s.V(3).Info("Found VPC with provided id", "id", vpcID)
+
+		requeue := true
+		if vpcDetails.Status != nil && *vpcDetails.Status == string(vpcv1.VPCStatusAvailableConst) {
+			requeue = false
+		}
+		s.SetResourceStatus(infrav1beta2.ResourceTypeVPC, &infrav1beta2.ResourceStatus{
+			ID:   *vpcID,
+			Name: vpcDetails.Name,
+			// Ready status will be invert of the need to requeue.
+			Ready: !requeue,
+		})
+
+		// After updating the Status of VPC, return with requeue or return as reconcile complete.
+		return requeue, nil
+	}
+
+	// If no VPC id was found, we need to create a new VPC.
+	s.V(3).Info("Creating a VPC")
+	vpcDetails, err := s.createVPC()
+	if err != nil {
+		return false, fmt.Errorf("failed to create vpc: %w", err)
+	}
+
+	s.V(3).Info("Successfully created VPC")
+	var vpcName *string
+	if vpcDetails != nil {
+		vpcName = vpcDetails.Name
+	}
+	s.SetResourceStatus(infrav1beta2.ResourceTypeVPC, &infrav1beta2.ResourceStatus{
+		ID:    *vpcDetails.ID,
+		Name:  vpcName,
+		Ready: false,
```

Comment:
I think IIUC in the next iteration you are expecting Ready to be set to true based on Status.